tools/autograd/derivatives.yaml

```diff
@@ -1775,6 +1775,19 @@
   target: smooth_l1_loss_backward(grad, target, self, reduction, beta)
   result: apply_loss_reduction(smooth_l1_loss_backward(self_t.conj(), self_p, target_p, at::Reduction::None, beta).conj() + smooth_l1_loss_backward(target_t.conj(), target_p, self_p, at::Reduction::None, beta).conj(), reduction)
 
+- name: margin_ranking_loss(Tensor input1, Tensor input2, Tensor target, float margin=0.0, int reduction=Mean) -> Tensor
+  input1, input2: "grad.defined() ? sub_backward(margin_ranking_loss_backward_input1(grad, input1, input2, target, margin, reduction)) : std::tuple<Tensor, Tensor>()"
+  target: "grad.defined() ? margin_ranking_loss_backward_target(grad, input1, input2, target, margin, reduction) : Tensor()"
```

Comment:
No need to check for grad defined here.

Reply:
could you explain why? i think this just follows the pattern that the other ops are using, so when is grad.defined necessary?

Reply:
See the notes at the top of `derivatives.yaml`. There's explained when we need to check for `grad.defined()`.

Reply:
okay, i've tried removing this and now grad tests fail. how exactly do i add undefined tensor support for this op? it seems i need to do it myself, is there a good example that i can follow?